src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -186,6 +186,15 @@ protected Signature(SignerInfo signerInfo, SignatureType type)
                     // This flag should only be set for verification scenarios, not signing.
                     chain.ChainPolicy.VerificationFlags = X509VerificationFlags.IgnoreNotTimeValid;
 
+                    if (settings.RevocationMode == RevocationMode.Offline)
+                    {
+                        chain.ChainPolicy.RevocationMode = X509RevocationMode.Offline;
+                    }
+                    else
+                    {
+                        chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+                    }
+
```

Comment:
I recommend moving this after the `SetCertBuildChainPolicy(...)` call, just in case it overrides the above revocation mode.